### DISCUSSION TITLE
最新值如何获取？有没有专用的函数或者sql语句啊

### DISCUSSION THREAD

**Discussion by skyformat99** [GreptimeTeam/greptimedb#4067] [Q&A]
### What problem does the new feature solve?

最新值如何获取？有没有专用的函数或者sql语句啊？

比如说，增加一个  last函数，类似于这样的：

select last(*) from tablename;

或者  select  last_row(*) from tablename;

可以参考 TaosDB啊，这个很方面。

### What does the feature do?

最新值如何获取？有没有专用的函数或者sql语句啊？

比如说，增加一个  last函数，类似于这样的：

select last(*) from tablename;

或者  select  last_row(*) from tablename;

可以参考 TaosDB啊，这个很方面。

**Comment by tisonkun** [comment]
感谢提问！这个属于使用问题，我先 move 到 Discussions 的 QA 上。稍后会请工程同事回复。

**Comment by waynexia** [comment]
Try `DISTINCT ON`:

```sql
select distinct on (host, idc) * from system_metrics order by host, idc, ts desc;
```

ref to https://docs.greptime.com/user-guide/query-data/sql#find-the-latest-data-of-time-series

**Reply by skyformat99** [reply]
可以是可以

> Try `DISTINCT ON`:
> 
> ```sql
> select distinct on (host, idc) * from system_metrics order by host, idc, ts desc;
> ```
> 
> ref to https://docs.greptime.com/user-guide/query-data/sql#find-the-latest-data-of-time-series

但是还是不那么简单方便啊；
别的好的 常用函数，可以追加支持一些。

比如  max  min   last   之类的

**Reply by tisonkun** [reply]
MAX 和 MIN 等聚合函数都是支持的 - https://docs.greptime.com/reference/sql/functions#datafusion-functions
